### PR TITLE
The Witness: Item loading refactor.

### PR DESCRIPTION
**Pull request by blastron**
## What is this fixing or adding?
This is an in-place refactor to the code that loads Witness-specific items from its configuration files. No actual AP- or player-facing behavior is different.

## How was this tested?
I generated a bunch of runs with fixed seed values and compared spoiler logs before/after to verify that they were identical.

## If this makes graphical changes, please attach screenshots.
N/A

### CODE REVIEW

**Comment by ThePhar** [comment]
@NewSoupVi

**Comment by NewSoupVi** [comment]
> @NewSoupVi

Blastron is a new dev for this world. I approved his changes and asked him to take the lead on the PR process as well :) So please treat him as the main person responsible~

**Comment by blastron** [comment]
thanks for adding the tag, i'll be sure to check those in the future

File: worlds/witness/__init__.py [hunk 1]
```diff
@@ -83,163 +103,87 @@ def generate_early(self):
...
+                    .place_locked_item(self.create_item(random_early_item))
+                if item_pool[random_early_item] == 1:
+                    item_pool.pop(random_early_item)
+                else:
+                    item_pool[random_early_item] -= 1
+
+        # Generate the actual items.
+        for item_name, quantity in item_pool.items():
+            self.multiworld.itempool += [self.create_item(item_name) for _ in range(0, quantity)]
+            if self.items.item_data[item_name].local_only:
```
Comment (blastron) [inline worlds/witness/__init__.py:182]: Note: this is not currently being used by anything live. I've tested it locally with my other changes.

**Comment by ThePhar** [comment]
> > @NewSoupVi
> 
> Blastron is a new dev for this world (and long-time dev of the client). I approved his changes and asked him to take the lead on the PR process as well :) So please treat him as the main person responsible~

Ah shoot, that's right I forgot. My bad. Should have checked my own CODEOWNERS sheet lol.

**Comment by black-sliver** [comment]
Sorry for merging the other PRs first, creating a merge conflict, but they were labelled as fixes.

@NewSoupVi can you still review and test, if not done already, and approve the changes (using the github approve feature)?

**Review by NewSoupVi — APPROVED** [review]
Already reviewed, I can approve yes

Altho yeah the merge conflicts will have to be solved ofc

**Review by ThePhar — APPROVED** [review]

**Pull request merged**